ui/src/vertical-geometry/util.ts

```diff
@@ -188,23 +176,47 @@ export function sumPaddings(p1: string, p2: string) {
     return parseFloat(p1) + parseFloat(p2);
 }
 
+function isLayoutGeometryItem(verticalGeometryItem: VerticalGeometryItem): boolean {
+    const layoutGeometryKeys: Array<keyof VerticalGeometryItem> = [
+        'layoutStartStation',
+        'layoutPointStation',
+        'layoutEndStation',
+    ];
+
+    return layoutGeometryKeys.some((layoutKey) => verticalGeometryItem[layoutKey] !== undefined);
+}
+
```

Comment:
Tähän taisi jäädä vielä vähän liikaa historian siipien havinaa minun bugisesta koodista, koska eikös tämän koko tarkistuksen voi lanata surutta pois?Jos bäkkäriltä jostain syystä valuu sellainen piste, jolla ei ole minkäänlaista tietoa siitä, mihin mikään osa siitä kuuluu paikannuspohjalla, niin sittenhän sitä vaan ei voida näyttää lainkaan täällä paikannuspohjan puolella.

substituteLayoutStationsForGeometryStations myös yksinkertaistuu tarkastuksen poiston myötä.

Reply:
Hmm, en saanut tästä ideasta ihan kiinni 🤔. 

Nykytilanteessahan tälle pystygeometrian kuvaajallehan pitää jotenkin saada tieto mitä kenttiä sen pitäisi käyttää riippuen piirtotilanteesta. Tällä hetkellä erotteluun käytetään tätä funktiota. Nuo .start/.point/.end-kentät ovat siis aina näillä VerticalGeometryItemeillä olemassa, mutta sisältävät virheellisiä (suunnitelmakontekstin) arvoja piirrettäessä sijaintiraidetta eli niitä ei silloin pitäisi käyttää (lainkaan).

Kuvaajalle voisi passata tiedon suunnitelman/sijantiraiteen piirrosta jo ylemmältä tasolta, jolloin piirtotilanteen tunnistusta ei tarvitsisi edes tehdä, vaan valittaisiin käytetyt kentät ylemmältä tasolta tulevan tiedon perusteella. Olisi varmaankin myös mahdollista korvata oikeat arvot noihin samoihin .start/.point/.end-kenttiin bäkkärillä, jos datan kaivamista tietyistä kentistä riippuen piirtotilanteesta ei koettaisi järkeväksi.

Tässä nykyversiossahan on myös mahdollinen bugi: Piirrettäessä sijaintiraiteen pystygeometriaa, jossa ainakin yksi VerticalGeometryItem sisältää yhden näistä .layout*-kentistä, mutta ainakin yksi VerticalGeometryItem ei sisällä ainuttakaan (koska niitä ei välttämättä oikeasti ole määritetty), niin piirtokoodi menee käyttämään suunnitelmapistettä, vaikkei sen pitäisi tällaisessa tilanteessa suunnitelmapistettä käyttää lainkaan. Tämänkin siis ratkaisisi jos piirtotilanteen tunnistaisi/valuttaisi alaspäin jostain ylemmältä tasolta, eikä sitä yrittäisi tunnistaa tässä yhteydessä jokaiselle VerticalGeometryItemille erikseen.

Reply:
Tämä tarpeeton tarkistus poistettu. Tähänhän osaan koodia ei siis edes päädytä piirrettäessä suunnitelman geometriaa, ks. `processLayoutGeometries()`.

Ei tainnut olla tarpeeksi tujut aamukahvit tuolloin ☕ 😄 .